core/src/main/java/org/lflang/generator/c/CGenerator.java

```diff
@@ -719,7 +722,22 @@ private void inspectReactorEResource(ReactorDecl reactor) {
               if (p.isPresent()) {
                 var property = p.get();
                 if (property.loadFromImport()) {
-                  property.update(this.targetConfig, pair, messageReporter);
+                  if (property instanceof FileListProperty) {
```

Comment:
I see why you did this (the base path is not available in `FileListProperty`), but it's kind of unclean to have this special case for handling `FileListProperty` here. Should we consider passing in a `GeneratorContext` in the `update` function instead, so we gain access to both the target config _and_ the file config?

Reply:
I also wanna do this. However, I see one problem here. Consider [DistributedCMakeIncludeSeparateCompile.lf](https://github.com/lf-lang/lingua-franca/blob/master/test/C/src/target/DistributedCMakeIncludeSeparateCompile.lf) looks like below.

```
target C {
  timeout: 0 sec
}

import Foo from "CMakeInclude.lf"
...
```
`CMakeInclude.lf`(https://github.com/lf-lang/lingua-franca/blob/master/test/C/src/target/CMakeInclude.lf) has the `cmake-include` property like below.

```
target C {
  cmake-include: [
      "../include/mlib-cmake-extension.cmake",
      "../include/foo-cmake-compile-definition.txt"],
  timeout: 0 sec
}
```

When we call `inspectReactorEResource` to gather `CMakeIinclude.lf`'s target property, we need to update the **parent reactor** (`DistributedCMakeIncludeSeparateCompile.lf`)'s `TargetConfig` using the **child reactor** (`CMakeInclude.lf`)'s `FileConfig`. Thus, solely passing one `GeneratorContext` doesn't resolve this problem. Do you think passing both parent _and_ child's `GeneratorContext` to the function `update` makes sense?

Reply:
Yep, you're right. I realized this too... Let me noodle on it a little bit. I have a long flight ahead, so I'll work on this stuff a bunch.

Reply:
I think we need an interface `ImportableTargetProperty` with an additional `update` method that carries an extra `sourceContext`. Only few implementations of `TargetProperty` will need to implement it.

Reply:
There already was a mechanism in place for federations that we could simply reuse, so I did that. I think this is good to merge now.